tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/jms/client/ConnectionTest.java

```diff
@@ -131,6 +131,27 @@ public void testTwoConnectionsSameIDThroughCF() throws Exception {
       session2.close();
    }
 
+   @Test
+   public void testTwoConnectionsSameIDThroughCFWithShareClientIDEnabeld() throws Exception {
+      ActiveMQConnectionFactory connectionFactory = new ActiveMQConnectionFactory("tcp://localhost:61616?clientID=myid;enableSharedClientID=true");
+
+      conn = connectionFactory.createConnection();
+      try {
+         conn2 = connectionFactory.createConnection();
+      } catch (InvalidClientIDException expected) {
+         Assert.fail("Should allow sharing of client IDs among the same CF");
+      }
+
+      Session session1 = conn.createSession();
+      Session session2 = conn.createSession();
```

Comment:
Probably should test conn2 as well here?

Reply:
Yes.  Copy and paste error ;).  Will fix.